packages/p2p-v2/src/pages/my-ads/screens/CreateEditAd/CreateEditAd.tsx

```diff
@@ -33,21 +34,28 @@ const CreateEditAd = () => {
             'min-order': '',
             'order-completion-time': `${orderPaymentPeriod ? (orderPaymentPeriod * 60).toString() : '3600'}`,
             'payment-method': '',
-            'preferred-countries': Object.keys(DUMMY_COUNTRIES),
+            'preferred-countries': Object.keys(countryList),
             'rate-value': '-0.01',
         },
         mode: 'all',
     });
 
+    const { getValues, handleSubmit, setValue } = methods;
+    useEffect(() => {
+        if (Object.keys(countryList).length > 0 && getValues('preferred-countries').length === 0) {
+            setValue('preferred-countries', Object.keys(countryList));
+        }
+    }, [countryList, getValues, setValue]);
+
     const shouldNotShowArchiveMessageAgain = localStorage.getItem('should_not_show_auto_archive_message_again');
-    const { getValues, handleSubmit } = methods;
     const onSubmit = () => {
         const payload = {
             amount: Number(getValues('amount')),
             eligible_countries: getValues('preferred-countries'),
             max_order_amount: Number(getValues('max-order')),
             min_order_amount: Number(getValues('min-order')),
             rate: Number(getValues('rate-value')),
+            rate_type: rateType as 'fixed' | 'float',
```

Comment:
can we change the typing in the useFloatingRate hook itself 👀 then in the future we don't need to keep using type assertions in other places